packages/compass-smoke-tests/src/cli.ts

```diff
@@ -201,12 +220,55 @@ async function run() {
   }
 }
 
+type AutoUpdateServerOptions = {
+  port: number;
+  allowDowngrades?: boolean;
+};
+
+function startAutoUpdateServer({
+  port,
+  allowDowngrades,
+}: AutoUpdateServerOptions) {
+  const env: Record<string, string> = {
+    ...process.env,
+    PORT: port.toString(),
+  };
+  if (allowDowngrades) {
+    env.UPDATE_CHECKER_ALLOW_DOWNGRADES = 'true';
+  }
+
+  // a git repo that is not published to npm that evergreen clones for us in CI
```

Comment:
I don't know how best to cleanly start this, especially in dev. This assumes that `compass-mongodb-com` is cloned right next to `compass` which is hacky.

My initial plan was to start the server using docker in evergreen so it is already running before we execute the smoke tests, but we don't have docker in every platform where we want to run the smoke tests, so I think we have to start the server manually.

Furthermore we'll also have two of these servers with different configuration that we'll need back-to-back because updating from the most recent release TO this test subject will require telling the auto-update server about this new unreleased copy of compass.